core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/queue/SchedulingDecisionMaker.scala

```diff
@@ -139,9 +139,9 @@ class SchedulingDecisionMaker(
           case (Running, Some(duration)) if staleActivationNum > 0 =>
             // we can safely get the value as we already checked the existence
             val containerThroughput = staleThreshold / duration
-            val num = ceiling(availableMsg.toDouble / containerThroughput)
+            val num = ceiling(staleActivationNum.toDouble / containerThroughput)
             // if it tries to create more containers than existing messages, we just create shortage
-            val actualNum = (if (num > availableMsg) availableMsg else num) - inProgress
+            val actualNum = (if (num > staleActivationNum) staleActivationNum else num) - inProgress
```

Comment:
I still believe we need to consider all available messages.
This is because there can be both stale activations and non-stale(but about to be staled) activations.
It takes some time to provision more containers. Even if we can add more containers based on stale activations but we would easily need to add more containers if the container throughput itself is not enough.
To avoid too much over-provision, I think we can just substitute `existing` as well.

``` 
val actualNum = (if (num > availableMsg) availableMsg else num) - inProgress - existing
```

Reply:
I think the code in my comment above covers this. I just pushed it so it's the code in the pr now. In the same decision interval, it will both make the calculation for additional containers needed for activations that are stale and increase containers based on # of available messages to meet the needed transactions per second. I think this is the most optimal case.